src/dsp_tools/fast_xmlupload/upload_files.py

```diff
@@ -151,51 +152,93 @@ def _check_upload_candidates(
     return True
 
 
+def _call_upload_without_processing_recursively(
+    file: Path,
+    sipi_url: str,
+    con: Connection,
+    err_msg: str,
+) -> bool:
+    """
+    There are different kinds of failures in _upload_without_processing(),
+    and each of them should be handled the same way:
+    By calling this function.
+
+    This function logs and prints the error message,
+    sleeps for 2 seconds,
+    and then calls _upload_without_processing() again.
+
+    Args:
+        file: file to upload
+        sipi_url: URL of the SIPI server
+        con: connection to the DSP server
+        err_msg: message to print and log
+
+    Returns:
+        return value of _upload_without_processing()
+    """
+    print(f"{datetime.now()}: ERROR: {err_msg}")
+    logger.error(err_msg, exc_info=True)
+    sleep(2)
```

Comment:
Why is this sleeping needed?

Reply:
In case the error has something to do with server overload, the server might need some time to recover. But I admit that this is hypothetical.